companion-code/syncing/rpc/src/main.rs

```diff
@@ -55,7 +57,9 @@ fn main() -> () {
         };
         print!("{} ", block.block_height());
 
-        wallet.apply_block_connected_to(&block.block, block.block_height(), block.connected_to()).unwrap();
+        wallet
+            .apply_block_connected_to(&block.block, block.block_height(), block.connected_to())
+            .unwrap();
```

Comment:
I'm tempted to rewrite this loop using a `while let`

```rust
while let Some(block) = emitter.next_block().unwrap() {
    print!("{} ", block.block_height());
    wallet
        .apply_block_connected_to(&block.block, block.block_height(), block.connected_to())
        .unwrap();
}
```

but I guess then we would lose the type annotation for block event `Option<BlockEvent<Block>>`

Reply:
I do like the while let Some() construct. How about simply adding a comment showing the type?

```rust
while let Some(block) = emitter.next_block().unwrap() {
    // the block variable is of type BlockEvent<Block>
    print!("{} ", block.block_height());
    wallet
        .apply_block_connected_to(&block.block, block.block_height(), block.connected_to())
        .unwrap();
}
```